lib/models/service.ts

```diff
@@ -0,0 +1,291 @@
+/*
+Copyright 2018 Balena
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import * as errors from 'balena-errors';
+import * as Promise from 'bluebird';
+import once = require('lodash/once');
+import {
+	PineOptionsFor,
+	Service,
+	ServiceEnvironmentVariable,
+} from '../../typings/balena-sdk';
+import { InjectedDependenciesParam, InjectedOptionsParam } from '../balena';
+import { findCallback, mergePineOptions } from '../util';
+
+const getServiceModel = (
+	deps: InjectedDependenciesParam,
+	opts: InjectedOptionsParam,
+) => {
+	const { pine } = deps;
+	const applicationModel = once(() => require('./application')(deps, opts));
+	const { buildDependentResource } = require('../util/dependent-resource');
+
+	const varModel = buildDependentResource(
+		{ pine },
+		{
+			resourceName: 'service_environment_variable',
+			resourceKeyField: 'name',
+			parentResourceName: 'service',
+			getResourceId: (id: number) => get(id, { $select: 'id' }).get('id'),
+			ResourceNotFoundError: errors.BalenaServiceNotFound,
+		},
+	);
+
+	exports = {};
+
+	// Not exported for now, but we could document & export in future
+	// if there are external use cases for this.
+	const get = (id: number, options: PineOptionsFor<Service>) => {
+		if (options == null) {
+			options = {};
+		}
+		return pine
+			.get({
+				resource: 'service',
+				id,
+				options,
+			})
+			.tap((service: Service | undefined) => {
+				if (service == null) {
+					throw new errors.BalenaServiceNotFound(id);
+				}
+			});
+	};
+
+	/**
+	 * @summary Get all services from an application
+	 * @name getAllByApplication
+	 * @public
+	 * @function
+	 * @memberof balena.models.service
+	 *
+	 * @param {String|Number} nameOrSlugOrId - application name (string), slug (string) or id (number)
+	 * @param {Object} [options={}] - extra pine options to use
+	 * @fulfil {Object[]} - services
+	 * @returns {Promise}
+	 *
+	 * @example
+	 * balena.models.service.getAllByApplication('MyApp').then(function(services) {
+	 * console.log(services);
```

Comment:
It seems that the indentation broke here, making the README examples to have zero indentation.
Let's re-indent them plz.
Applies to the two other examples right below.

Reply:
the linter was complaining about the original formatting, but I managed to get it fixed now.